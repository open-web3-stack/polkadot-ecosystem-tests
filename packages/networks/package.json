{
	"name": "@e2e-test/networks",
	"type": "module",
	"dependencies": {
		"@acala-network/chopsticks-testing": "^0.13.1",
		"@e2e-test/shared": "workspace:*"
	},
	"main": "./dist/cjs/index.js",
	"module": "./dist/esm/index.js",
	"files": [
		"*",
		"!tsconfig.json",
		"!**/*.test.ts"
	],
	"exports": {
		".": {
			"require": "./dist/cjs/index.js",
			"import": "./dist/esm/index.js",
			"default": "./dist/esm/index.js"
		},
		"./chains": {
			"require": "./dist/cjs/chains/index.js",
			"import": "./dist/esm/chains/index.js",
			"default": "./dist/esm/chains/index.js"
		},
		"./*": {
			"require": "./dist/cjs/*.js",
			"import": "./dist/esm/*.js",
			"default": "./dist/esm/*.js"
		}
	}
}
