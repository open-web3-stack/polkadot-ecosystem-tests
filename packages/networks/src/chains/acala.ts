import { defaultAccount } from '@e2e-test/shared'

import { defineChain } from '../defineChain.js'

const custom = {
  acala: {
    relayToken: 'DOT',
    relayLiquidToken: 'LDOT',
    stableToken: 'AUSD',

    paraAccount: '13YMK2eYoAvStnzReuxBjMrAvPXmmdsURwZvc62PrdXimbNy',
    dot: { Token: 'DOT' },
    ldot: { Token: 'LDOT' },
    dai: { Erc20: '0x54a37a01cd75b616d63e0ab665bffdb0143c52ae' },
    wbtc: { ForeignAsset: 5 },
    ausd: { Token: 'AUSD' },
    aca: { Token: 'ACA' },
    lcdot: { LiquidCrowdloan: 13 },
  },
  karura: {
    relayToken: 'KSM',
    relayLiquidToken: 'LKSM',
    stableToken: 'KUSD',

    paraAccount: '13YMK2eYoAvStnzReuxBjMrAvPXmmdsURwZvc62PrdXimbNy',
    ksm: { Token: 'KSM' },
    lksm: { Token: 'LKSM' },
    usdt: { ForeignAsset: 7 },
    rmrk: { ForeignAsset: 0 },
    dai: { Erc20: '0x4bb6afb5fa2b07a5d1c499e1c3ddb5a15e709a71' },
    ausd: { Token: 'KUSD' },
    kar: { Token: 'KAR' },
  },
}

const getInitStorages = (config: typeof custom.acala | typeof custom.karura) => ({
  System: {
    account: [[[defaultAccount.alice.address], { providers: 4, data: { free: 10 * 1e12 } }]],
  },
  Tokens: {
    accounts: [
      [[defaultAccount.alice.address, { Token: config.relayToken }], { free: 10 * 1e12 }],
      [[defaultAccount.alice.address, { Token: config.relayLiquidToken }], { free: 100 * 1e12 }],
      [[defaultAccount.alice.address, { Token: config.stableToken }], { free: 1000 * 1e12 }],
    ],
  },
  Sudo: {
    key: defaultAccount.alice.address,
  },
  EvmAccounts: {
    accounts: [[['0x82a258cb20e2adb4788153cd5eb5839615ece9a0'], defaultAccount.alice.address]],
    evmAddresses: [[[defaultAccount.alice.address], '0x82a258cb20e2adb4788153cd5eb5839615ece9a0']],
  },
  Homa: {
    // avoid impact test outcome
    $removePrefix: ['redeemRequests', 'unbondings', 'toBondPool'],
    // so that bump era won't trigger unbond
    relayChainCurrentEra: 100,
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers', 'supportedVersion'],
  },
})

export const acala = defineChain({
  name: 'acala',
  endpoint: 'wss://acala-rpc.aca-api.network',
  paraId: 2000,
  custom: custom.acala,
  initStorages: getInitStorages(custom.acala),
})

export const karura = defineChain({
  name: 'karura',
  endpoint: 'wss://karura-rpc.aca-api.network',
  paraId: 2000,
  custom: custom.karura,
  initStorages: getInitStorages(custom.karura),
})
