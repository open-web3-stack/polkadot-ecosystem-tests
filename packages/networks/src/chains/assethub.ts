import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  assetHubPolkadot: {
    dot: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
  assetHubKusama: {
    ksm: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
}

const getInitStorages = (config: typeof custom.assetHubPolkadot | typeof custom.assetHubKusama) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }]],
  },
  Assets: {
    account: [
      [[config.usdtIndex, defaultAccounts.alice.address], { balance: 1000e6 }], // USDT
    ],
  },
})

export const assetHubPolkadot = defineChain({
  name: 'assetHubPolkadot',
  endpoint: 'wss://rpc-asset-hub-polkadot.luckyfriday.io',
  paraId: 1000,
  custom: custom.assetHubPolkadot,
  initStorages: getInitStorages(custom.assetHubPolkadot),
})

export const assetHubKusama = defineChain({
  name: 'assetHubKusama',
  endpoint: 'wss://rpc-asset-hub-kusama.luckyfriday.io',
  paraId: 1000,
  custom: custom.assetHubKusama,
  initStorages: getInitStorages(custom.assetHubKusama),
})
