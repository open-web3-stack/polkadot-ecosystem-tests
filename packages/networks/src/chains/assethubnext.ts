import { defaultAccounts, defaultAccountsSr25519 } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  assetHubNextWestend: {
    wnd: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
}

const getInitStorages = (config: typeof custom.assetHubNextWestend) => ({
  System: {
    account: [
      [[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }],
      [[defaultAccountsSr25519.alice.address], { providers: 1, data: { free: 1000e10 } }],
    ],
  },
  Assets: {
    account: [
      [[config.usdtIndex, defaultAccounts.alice.address], { balance: 1000e6 }], // USDT
    ],
  },
})

export const assetHubNextWestend = defineChain({
  name: 'assetHubNextWestend',
  endpoint: 'wss://westend-asset-hub-next-rpc.parity-chains-scw.parity.io',
  paraId: 1100,
  custom: custom.assetHubNextWestend,
  initStorages: getInitStorages(custom.assetHubNextWestend),
})
