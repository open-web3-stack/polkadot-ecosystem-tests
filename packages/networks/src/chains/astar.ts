import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  astar: {
    relayToken: '340282366920938463463374607431768211455',
    aUSDToken: '18446744073709551617',

    paraAccount: '13YMK2eZzuFY1WZGagpYtTgbWBWGdoUD2CtrPj1mQPjY8Ldc',
    dot: 340282366920938463463374607431768211455n,
    astr: { Concrete: { parents: 0, interior: 'Here' } },
    aca: 18446744073709551616n,
    usdt: 4294969280n,
  },
  shiden: {
    relayToken: '340282366920938463463374607431768211455',
    aUSDToken: '18446744073709551616',

    paraAccount: 'F7fq1jNy74AqkJ1DP4KqSrWtnTGtXfNVoDwFhTvvPxUvJaq',
    ksm: 340282366920938463463374607431768211455n,
    sdn: { Concrete: { parents: 0, interior: 'Here' } },
    kar: 18446744073709551618n,
    usdt: 4294969280n,
  },
}

const getInitStorages = (config: typeof custom.astar | typeof custom.shiden) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: '100000000000000000000' } }]],
  },
  Assets: {
    account: [
      [[config.relayToken, defaultAccounts.alice.address], { balance: 10 * 1e12 }],
      [[config.aUSDToken, defaultAccounts.alice.address], { balance: 10 * 1e12 }],
    ],
  },
  Sudo: {
    key: defaultAccounts.alice.address,
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers', 'supportedVersion'],
  },
})

export const astar = defineChain({
  name: 'astar',
  paraId: 2006,
  endpoint: 'wss://astar-rpc.dwellir.com',
  custom: custom.astar,
  initStorages: getInitStorages(custom.astar),
})

export const shiden = defineChain({
  name: 'shiden',
  paraId: 2007,
  endpoint: 'wss://shiden-rpc.dwellir.com',
  custom: custom.shiden,
  initStorages: getInitStorages(custom.shiden),
})
