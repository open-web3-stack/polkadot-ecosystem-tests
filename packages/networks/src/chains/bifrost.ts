import { defineChain } from '../defineChain.js'
import { defaultAccounts, defaultAccountsSr25519, testAccounts } from '../testAccounts.js'

const custom = {
  bifrostPolkadot: {
    relayToken: 'DOT',
    bnc: { Native: 'BNC' },
  },
  bifrostKusama: {
    relayToken: 'KSM',
    bnc: { Token: 'BNC' },
  },
}

const getInitStorages = (_config: typeof custom.bifrostPolkadot | typeof custom.bifrostKusama) => ({
  System: {
    account: [
      [[defaultAccounts.alice.address], { providers: 1, data: { free: 666e12 } }],
      [[defaultAccountsSr25519.alice.address], { providers: 1, data: { free: 666e12 } }],
      [[testAccounts.alice.address], { providers: 1, data: { free: 666e12 } }],
    ],
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers'],
  },
})

export const bifrostPolkadot = defineChain({
  name: 'bifrostPolkadot',
  endpoint: 'wss://bifrost-polkadot.ibp.network',
  paraId: 2030,
  custom: custom.bifrostPolkadot,
  initStorages: getInitStorages(custom.bifrostPolkadot),
})

export const bifrostKusama = defineChain({
  name: 'bifrostKusama',
  endpoint: 'wss://us.bifrost-rpc.liebi.com/ws',
  paraId: 2001,
  custom: custom.bifrostKusama,
  initStorages: getInitStorages(custom.bifrostKusama),
})
