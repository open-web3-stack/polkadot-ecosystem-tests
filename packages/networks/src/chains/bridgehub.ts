import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  bridgeHubPolkadot: {
    dot: { Concrete: { parents: 1, interior: 'Here' } },
  },
  bridgeHubKusama: {
    ksm: { Concrete: { parents: 1, interior: 'Here' } },
  },
}

const getInitStorages = (_config: typeof custom.bridgeHubPolkadot | typeof custom.bridgeHubKusama) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }]],
  },
})

export const bridgeHubPolkadot = defineChain({
  name: 'bridgeHubPolkadot',
  endpoint: 'wss://polkadot-bridge-hub-rpc.polkadot.io',
  paraId: 1002,
  custom: custom.bridgeHubPolkadot,
  initStorages: getInitStorages(custom.bridgeHubPolkadot),
})

export const bridgeHubKusama = defineChain({
  name: 'bridgeHubKusama',
  endpoint: 'wss://kusama-bridge-hub-rpc.polkadot.io',
  paraId: 1002,
  custom: custom.bridgeHubKusama,
  initStorages: getInitStorages(custom.bridgeHubKusama),
})
