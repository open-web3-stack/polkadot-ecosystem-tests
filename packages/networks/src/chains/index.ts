export * from './acala.js'
export * from './assethub.js'
export * from './astar.js'
export * from './hydraDX.js'
export * from './moonbeam.js'
export * from './polkadot.js'
