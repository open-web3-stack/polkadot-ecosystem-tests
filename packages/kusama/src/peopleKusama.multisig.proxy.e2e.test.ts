import { peopleKusama } from '@e2e-test/networks/chains'
import { baseMultisigProxyE2Etests, type ParaTestConfig, PeopleProxyTypes, registerTestTree } from '@e2e-test/shared'

const testConfig: ParaTestConfig = {
  testSuiteName: 'People Kusama Multisig Proxy',
  addressEncoding: 2,
  blockProvider: 'Local',
  asyncBacking: 'Disabled',
}

registerTestTree(baseMultisigProxyE2Etests(peopleKusama, testConfig, PeopleProxyTypes))
