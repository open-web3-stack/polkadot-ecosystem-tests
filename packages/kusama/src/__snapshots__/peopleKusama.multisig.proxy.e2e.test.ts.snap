// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`People Kusama Multisig Proxy > failure tests > 2-of-3 multisig as standard proxy (non-transfer) > events when Alice creates multisig 1`] = `
[
  {
    "data": {
      "approving": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "callHash": "(hash)",
      "multisig": "EF9xmEeFv3nNVM3HyLAMTV5TU8jua5FRXCE116yfbbrZbCL",
    },
    "method": "NewMultisig",
    "section": "multisig",
  },
]
`;

exports[`People Kusama Multisig Proxy > failure tests > 2-of-3 multisig as standard proxy (non-transfer) > events when Bob approves multisig call 1`] = `
[
  {
    "data": {
      "approving": "FoQJpPyadYccjavVdTWxpxU7rUEaYhfLCPwXgkfD6Zat9QP",
      "callHash": "(hash)",
      "multisig": "EF9xmEeFv3nNVM3HyLAMTV5TU8jua5FRXCE116yfbbrZbCL",
      "result": "Ok",
      "timepoint": {
        "height": "(redacted)",
        "index": 2,
      },
    },
    "method": "MultisigExecuted",
    "section": "multisig",
  },
]
`;

exports[`People Kusama Multisig Proxy > success tests > 2-of-3 multisig as standard proxy (any) > events when Alice creates multisig 1`] = `
[
  {
    "data": {
      "approving": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "callHash": "(hash)",
      "multisig": "EF9xmEeFv3nNVM3HyLAMTV5TU8jua5FRXCE116yfbbrZbCL",
    },
    "method": "NewMultisig",
    "section": "multisig",
  },
]
`;

exports[`People Kusama Multisig Proxy > success tests > 2-of-3 multisig as standard proxy (any) > events when Bob approves multisig call 1`] = `
[
  {
    "data": {
      "approving": "FoQJpPyadYccjavVdTWxpxU7rUEaYhfLCPwXgkfD6Zat9QP",
      "callHash": "(hash)",
      "multisig": "EF9xmEeFv3nNVM3HyLAMTV5TU8jua5FRXCE116yfbbrZbCL",
      "result": "Ok",
      "timepoint": {
        "height": "(redacted)",
        "index": 2,
      },
    },
    "method": "MultisigExecuted",
    "section": "multisig",
  },
]
`;

exports[`People Kusama Multisig Proxy > success tests > 2-of-3 multisig as standard proxy with announcement and rejection > events when Bob approves multisig call 1`] = `
[
  {
    "data": {
      "approving": "FoQJpPyadYccjavVdTWxpxU7rUEaYhfLCPwXgkfD6Zat9QP",
      "callHash": "(hash)",
      "multisig": "EF9xmEeFv3nNVM3HyLAMTV5TU8jua5FRXCE116yfbbrZbCL",
      "result": "Ok",
      "timepoint": {
        "height": "(redacted)",
        "index": 2,
      },
    },
    "method": "MultisigExecuted",
    "section": "multisig",
  },
]
`;

exports[`People Kusama Multisig Proxy > success tests > 2-of-3 multisig as standard proxy with announcement and rejection > events when Dave rejects the announced call 1`] = `[]`;

exports[`People Kusama Multisig Proxy > success tests > 2-of-3 multisig with pure proxy (any) > events when Alice creates multisig 1`] = `
[
  {
    "data": {
      "approving": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "callHash": "(hash)",
      "multisig": "HjV6E4bhFtVYVHtFGcvaKoPjMei1yQDRicZ5YZVZPBj4LFf",
    },
    "method": "NewMultisig",
    "section": "multisig",
  },
]
`;

exports[`People Kusama Multisig Proxy > success tests > 2-of-3 multisig with pure proxy (any) > events when Bob's proxy approves the multisig call 1`] = `
[
  {
    "data": {
      "approving": "EyCT2L2XcppNEvDXVuiHxHwwV5ntWiJGRC4yByG8yktdFLT",
      "callHash": "(hash)",
      "multisig": "HjV6E4bhFtVYVHtFGcvaKoPjMei1yQDRicZ5YZVZPBj4LFf",
      "result": "Ok",
      "timepoint": {
        "height": "(redacted)",
        "index": 2,
      },
    },
    "method": "MultisigExecuted",
    "section": "multisig",
  },
]
`;

exports[`People Kusama Multisig Proxy > success tests > 2-of-3 multisig with pure proxy (non-transfer) > events when Alice creates multisig 1`] = `
[
  {
    "data": {
      "approving": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "callHash": "(hash)",
      "multisig": "E7yySYCVFxjKjZFHvEocrjiGWgRRX4Lr2TC98XXesf9eEm2",
    },
    "method": "NewMultisig",
    "section": "multisig",
  },
]
`;

exports[`People Kusama Multisig Proxy > success tests > 2-of-3 multisig with pure proxy (non-transfer) > events when Bob's proxy approves the multisig call 1`] = `
[
  {
    "data": {
      "approving": "HZAMcszveFWnZcqBkaHJTTyWXYfCqCF88maVMtBhPRvFfVw",
      "callHash": "(hash)",
      "multisig": "E7yySYCVFxjKjZFHvEocrjiGWgRRX4Lr2TC98XXesf9eEm2",
      "result": "Ok",
      "timepoint": {
        "height": "(redacted)",
        "index": 2,
      },
    },
    "method": "MultisigExecuted",
    "section": "multisig",
  },
]
`;

exports[`People Kusama Multisig Proxy > success tests > 2-of-3 multisig with pure proxy multisig > events when Alice creates multisig 1`] = `
[
  {
    "data": {
      "approving": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "callHash": "(hash)",
      "multisig": "DN8G6LsDKbU8TPDS6WGEGWTSTjhGRtCZ3BBjqr2MVS4SSkD",
    },
    "method": "NewMultisig",
    "section": "multisig",
  },
]
`;

exports[`People Kusama Multisig Proxy > success tests > Cancel 2-of-3 multisig with pure proxy before any other approvals > events when Alice creates multisig 1`] = `
[
  {
    "data": {
      "approving": "Gs8KxaM3NEWQ63FhnxmdSMnp2KpEEtPee6TzBSAmLm95nAx",
      "callHash": "(hash)",
      "multisig": "GWjoiJaRBwJy9t5Sf994YpuHyjf1FHrFZ2tnFxTZaZejue9",
    },
    "method": "NewMultisig",
    "section": "multisig",
  },
]
`;
